sycl/include/CL/sycl/detail/generic_type_traits.hpp

```diff
@@ -249,7 +253,11 @@ using is_longn = typename is_contained<
 // genlong: long int, longn
 template <typename T>
 using is_genlong =
-    std::integral_constant<bool, is_contained<T, type_list<cl_long>>::value ||
+    std::integral_constant<bool, is_contained<T, type_list<
+#if _MSC_VER
```

Comment:
Melanie, I don't quite understand the problem and possible solutions here. I only heard there were at least two solutions, right? So, I would rather rely on other reviewers in this PR.

My question here is: why you used _MSC_VER to guard this code? I.e. not _WIN32.
Also, what happens if you do not guard those lines at all, i.e. have/use same code for Windows & Linux?

Reply:
Melanie said: If we didn't use conditional compilation, then we'd be getting similar name resolution compfail problems on Linux.

Are you sure? Long or cl_long should just be long or long on Linux.